test/qunit/unit-es5/test-component-scheduler.js

```diff
@@ -165,6 +165,85 @@ test('disable component during onEnable', function () {
     childComp.onEnable.once('could re-enable component');
 });
 
+test('change parent during onDisable', function () {
+   
+    var rootNode , preNode , testNode , child , childComp;
+
+    var vendorError = cc.errorID;
+    cc.errorID = new Callback().enable();
+
+    //Pattern one : the Hierarchy change during onDisable 
+
+    var nodes1 = createNodes({
+        preNode: {},
+        testNode: {         
+            comps: cc.Component,
+        },
+    });
+    rootNode = nodes1.root;
+    preNode = nodes1.preNode;
+    testNode = nodes1.testNode;
+    testNodeComp = nodes1.testNodeComps[0];
+
+    testNodeComp.onDisable = function() {
+        testNode.parent = preNode;
+    };
+    
+    nodes1.attachToScene();
+
+    testNode.active = false;
+    cc.errorID.once('Should not change owned Hierarchy during deactivating');
+
+    //Pattern two : the Hierarchy change during onDisable 
+
+    var nodes2 = createNodes({
+        preNode: {},
+        testNode: {
+            comps: cc.Component,
+        },
+    });
+    rootNode = nodes2.root;
+    preNode = nodes2.preNode;
+    testNode = nodes2.testNode;
+    testNodeComp = nodes2.testNodeComps[0];
+
+    testNodeComp.onDisable = function() {
+        preNode.parent = testNode;
+    }
+
+    nodes2.attachToScene();
+
+    testNode.active = false; 
+    cc.errorID.once('Should not change the others\' Hierarchy to any deactivating node');
```

Comment:
Should report error when append new child to a target node during its deactivation